docs/configuration/http_conn_man/route_config/vhost.rst

```diff
@@ -27,10 +27,10 @@ name
 
 domains
   *(required, array)* A list of domains (host/authority header) that will be matched to this
-  virtual host. Currently, wildcard matching of the form "\*.foo.com" is not supported, however
-  a special entry "\*" is allowed which will match any host/authority header. Only a single virtual
-  host in the entire route configuration can match on "\*". A domain must be unique across all
-  virtual hosts or the config will fail to load.
+  virtual host. Wildcard hosts are supported in the form of "\*.foo.com" or "\*-bar.foo.com".
+  Additionally, a special entry "\*" is allowed which will match any host/authority header.
```

Comment:
@tschroed given that we use glob semantics for *, I suggest adding another line about the semantics as well. Something alone the lines of "Note: In *-bar.foo.com, * matches any non-empty string, i.e. baz-bar.foo.com but not -bar.foo.com. A separate virtual host entry must be used to match -bar.foo.com